python/tank/descriptor/io_descriptor/appstore.py

```diff
@@ -667,18 +668,24 @@ def __get_app_store_proxy_setting(self):
         Retrieve the app store proxy settings. If the key app_store_http_proxy is not found in the
         ``shotgun.yml`` file, the proxy settings from the client site connection will be used. If the key
         is found, than its value will be used. Note that if the ``app_store_http_proxy`` setting is set
-        to ``null`` in the configuration file, it means that the app store proxy is being forced to ``None``
+        to ``null`` or an empty string in the configuration file, it means that the app store proxy is being forced to ``None``
         and therefore won't be inherited from the http proxy setting.
 
         :returns: The http proxy connection string.
         """
         config_data = shotgun.get_associated_sg_config_data()
         if config_data and constants.APP_STORE_HTTP_PROXY in config_data:
-            return config_data[constants.APP_STORE_HTTP_PROXY]
-        else:
-            # Use the http proxy from the connection so we don't have to run
-            # the connection hook again.
-            return self._sg_connection.config.raw_http_proxy
+            # Cast any falsy value (e.g. "") from the settings into None.
+            return config_data[constants.APP_STORE_HTTP_PROXY] or None
+
+        settings = UserSettings()
+        if settings.is_default_app_store_http_proxy_set():
```

Comment:
this bit feels like leaky abstraction to me. Can you see any downside to having all the lofic inside the `UserSettings` instead and replace all the checks and logic with a

```
return UserSettings().app_store_proxy
```